cpp/include/cudf/scalar/scalar.hpp

```diff
@@ -377,6 +391,7 @@ class fixed_point_scalar : public scalar {
 
  protected:
   rmm::device_scalar<rep_type> _data{};  ///< device memory containing the value
+  rmm::device_scalar<int32_t> _scale{};  ///< device memory containing the scale
```

Comment:
Seems we are storing the scale in this class twice. Once in the base class's `_data_type` and once here. I just worry they could get out synch. Perhaps just create an accessor in this class that reads it out of the protected `_data_type` member?

Reply:
Yea, honestly I have no I idea what I was thinking here at the time. I should have been just accessing the value in the base `scalar` class. Fix incoming.